### PR TITLE
PLANNER-1517: Add Admin page

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
Depends on #271

### CODE REVIEW

File: employee-rostering-frontend/src/store/roster/operations.ts [hunk 1]
```diff
@@ -85,201 +90,245 @@ const updateInterval = 1000;
...
-      const solvingLength = 30 * 1000;
-      dispatch(actions.solveRoster());
-      dispatch(alert.showInfoMessage("startSolvingRoster", {
-        startSolvingTime: moment(solvingStartTime).format("LLL")
+export const solveRoster:
+ThunkCommandFactory<void, AddAlertAction | SolveRosterAction> = () => (dispatch, state, client) => {
+  const tenantId = state().tenantData.currentTenantId;
+  return client.post(`/tenant/${tenantId}/roster/solve`, {}).then(() => {
+    const solvingStartTime: number = new Date().getTime();
+    const solvingLength = 30 * 1000;
```
[the remark this answers is not in the document]
Comment (Christopher-Chianelli) [reply employee-rostering-frontend/src/store/roster/operations.ts]: Comes from the solver configuration. Perhaps `solve` endpoint can be modified to provide when solving will stop. Hard to share the constant between both frontend and backend. The other option is to repeating query the solver status endpoint.

File: employee-rostering-frontend/src/ui/pages/admin/NewTenantFormModal.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  onClose: () => void;
+}
+
+export type Props = StateProps & DispatchProps & OwnProps;
+
+export function isFormCompleted(rs: Partial<RosterState>): rs is RosterState {
+  return rs.draftLength !== undefined && rs.firstDraftDate !== undefined &&
+    rs.lastHistoricDate !== undefined && rs.publishLength !== undefined &&
+    rs.publishNotice !== undefined && rs.rotationLength !== undefined &&
+    rs.tenant !== undefined && rs.timeZone !== undefined &&
```
[the remark this answers is not in the document]
Comment (Christopher-Chianelli) [reply employee-rostering-frontend/src/ui/pages/admin/NewTenantFormModal.tsx]: No; will try to fix

**Comment by Christopher-Chianelli** [comment]
@yurloc Proper validation for all components (New Shift/etc. dialogs + table) will be handled in a new PR.

File: employee-rostering-frontend/src/ui/pages/admin/NewTenantFormModal.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+          >
+            Close
+          </Button>
+        ),
+        (
+          <Button
+            aria-label="Save"
+            key={2}
+            onClick={() => {
+              if (isFormCompleted(formData)) {
```
[the remark this answers is not in the document]
Comment (Christopher-Chianelli) [reply employee-rostering-frontend/src/ui/pages/admin/NewTenantFormModal.tsx]: Done (button is disabled until form passes validation)

File: employee-rostering-frontend/src/ui/pages/admin/NewTenantFormModal.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Christopher-Chianelli) [reply employee-rostering-frontend/src/ui/pages/admin/NewTenantFormModal.tsx]: Fixed

File: employee-rostering-frontend/src/store/roster/operations.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (yurloc) [reply employee-rostering-frontend/src/store/roster/operations.ts]: Thanks for confirmation. This is out of scope of this PR. I've filed #280.

**Comment by Christopher-Chianelli** [comment]
@yurloc Implemented your suggestions; full validation and a better solution for detecting end of solving is out of scope for this PR; will be addressed in a different PR.

**Comment by yurloc** [comment]
Yes, totally agree. The issues you mentioned are out of scope.

**Pull request merged**